ArcGISRuntimeSDKQt_QMLSamples/Analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.qml

```diff
@@ -0,0 +1,423 @@
+        }
+    }
+
+    function comparisonOperatorToString(comparisonOperator) {
+        switch (comparisonOperator) {
+        case Enums.UtilityAttributeComparisonOperatorEqual:
+            return "Equal";
+        case Enums.UtilityAttributeComparisonOperatorNotEqual:
+            return "NotEqual";
+        case Enums.UtilityAttributeComparisonOperatorGreaterThan:
+            return "GreaterThan";
+        case Enums.UtilityAttributeComparisonOperatorGreaterThanEqual:
+            return "GreaterThanEqual";
+        case Enums.UtilityAttributeComparisonOperatorLessThan:
+            return "LessThan";
+        case Enums.UtilityAttributeComparisonOperatorLessThanEqual:
+            return "LessThanEqual";
+        case Enums.UtilityAttributeComparisonOperatorIncludesTheValues:
+            return "IncludesTheValues";
+        case Enums.UtilityAttributeComparisonOperatorDoesNotIncludeTheValues:
+            return "DoesNotIncludeTheValues";
+        case Enums.UtilityAttributeComparisonOperatorIncludesAny:
+            return "IncludesAny";
+        case Enums.UtilityAttributeComparisonOperatorDoesNotIncludeAny:
+            return "DoesNotIncludeAny";
+        default:
+            return null;
+        }
+    }
+
+    function addCondition() {
+        // NOTE: You may also create a UtilityCategoryComparison with UtilityNetworkDefinition.Categories and UtilityCategoryComparisonOperator.
+
+        if (traceConfiguration === null) {
+            traceConfiguration = ArcGISRuntimeEnvironment.createObject("UtilityTraceConfiguration");
+        }
+        if (traceConfiguration.traversability === null) {
+            traceConfiguration.traversability = ArcGISRuntimeEnvironment.createObject("UtilityTraversability");
+        }
+
+        const selectedNetworkAttribute = networkDefinition.networkAttribute(networkAttributeComboBox.currentText);
+        const data = convertToDataType(valueSelectionComboBox.visible ? valueSelectionComboBox.currentIndex : inputTextField.text, selectedNetworkAttribute.dataType)
+
+        if (isNaN(data)) {
+            dialogText.text = qsTr("Invalid input. Please try again.")
+            dialog.visible = true;
+            return;
+        }
+
+        // NOTE: You may also create a UtilityNetworkAttributeComparison with another NetworkAttribute.
+        const expression = ArcGISRuntimeEnvironment.createObject("UtilityNetworkAttributeComparison", {
+                                                                     networkAttribute: selectedNetworkAttribute,
+                                                                     comparisonOperator: comparisonOperatorComboBox.currentIndex,
+                                                                     value: data});
+
+        const otherExpression = traceConfiguration.traversability.barriers;
+
+        if (otherExpression) {
```

Comment:
Seems confusing to read this. `otherExpression` = `barriers`, which sounds like a list, and a list cannot be null.

Reply:
I agree that is definitely confusing. I could add a comment explaining that this is checking to make sure that this is not the first expression being added to the barriers condition.

After thinking about this a little, I think it would be better to remove this though. I had added some code here and there in the case the user adds in their own data. Our general stance on this is that the sample should work as is and if the user changes the data, it's expected they could modify the code as well, if needed. This checks for an edge case that isn't expected in this sample.

Reply:
That would be best if you can remove this. I wouldn't expect the sample code to handle any data. It is tuned for the dataset it is written against.